src/main/java/emissary/util/DateTimeFormatParser.java

```diff
@@ -0,0 +1,211 @@
+            configG = emissary.config.ConfigUtil.getConfigInfo(DateTimeFormatParser.class);
+        } catch (IOException e) {
+            logger.error("Cannot open default config file", e);
+            return;
+        }
+
+        for (final String dfentry : configG.findEntries("DATE_TIME_ZONE_FORMAT")) {
+            try {
+                final DateTimeFormatter sdf =
+                        new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(dfentry).toFormatter().withZone(GMT);
+                dateTimeZoneFormats.add(sdf);
+            } catch (Exception ex) {
+                logger.debug("DATE_TIME_ZONE_FORMAT entry '{}' cannot be parsed", dfentry, ex);
+            }
+        }
+        logger.debug("Loaded {} DATE_TIME_ZONE_FORMAT entries", dateTimeZoneFormats.size());
+        for (final String dfentry : configG.findEntries("DATE_TIME_OFFSET_FORMAT")) {
+            try {
+                final DateTimeFormatter sdf =
+                        new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(dfentry).toFormatter();
+                dateTimeOffsetFormats.add(sdf);
+            } catch (Exception ex) {
+                logger.debug("DATE_TIME_OFFSET_FORMAT entry '{}' cannot be parsed", dfentry, ex);
+            }
+        }
+        logger.debug("Loaded {} DATE_TIME_OFFSET_FORMAT entries", dateFormats.size());
+        for (final String dfentry : configG.findEntries("DATE_TIME_FORMAT")) {
+            try {
+                final DateTimeFormatter sdf =
+                        new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(dfentry).toFormatter();
+                dateTimeFormats.add(sdf);
+            } catch (Exception ex) {
+                logger.debug("DATE_TIME_FORMAT entry '{}' cannot be parsed", dfentry, ex);
+            }
+        }
+        logger.debug("Loaded {} DATE_TIME_FORMAT entries", dateTimeFormats.size());
+
+
+        for (final String dfentry : configG.findEntries("DATE_FORMAT")) {
+            try {
+                final DateTimeFormatter sdf =
+                        new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(dfentry).toFormatter();
+                dateFormats.add(sdf);
+            } catch (Exception ex) {
+                logger.debug("DATE_FORMAT entry '{}' cannot be parsed", dfentry, ex);
+            }
+        }
+    }
+
+    /**
+     * Parse an RFC-822 Date or one of the thousands of variants make a quick attempt to normalize the timezone information
+     * and get the timestamp in GMT. Should change to pass in a default from the U124 header
+     *
+     * @param dateString the string date from the RFC 822 Date header
+     * @param supplyDefaultOnBad when true use current date if sentDate is unparseable
+     * @return the GMT time of the event or NOW if unparseable, or null if supplyDefaultOnBad is false
+     */
+    public static LocalDateTime parseDate(final String dateString, final boolean supplyDefaultOnBad) {
```

Comment:
I was trying to avoid having to check for date == null more times than necessary and maybe add some readability. I don't know if this does anything for readability, but I'll share it anyway.
```suggestion
    public static LocalDateTime parseDate(final String dateString, final boolean supplyDefaultOnBad) {
```
    public static LocalDateTime parseDate(final String dateString, final boolean supplyDefaultOnBad) {
        LocalDateTime date = null;

        if (dateString != null && dateString.length() > 0) {
            // Take it apart and stick it back together to
            // get rid of multiple contiguous spaces
            String instr = cleanDateString(dateString);

            for (String entry : entrytypes) {
                switch (entry) {
                    case "DATE_TIME_ZONE_FORMAT":
                        date = tryParseWithDateTimeZoneFormats(instr);
                        break;
                    case "DATE_TIME_OFFSET_FORMAT":
                        date = tryParseWithDateTimeOffsetFormats(instr);
                        break;
                    case "DATE_TIME_FORMAT":
                        date = tryParseWithDateTimeFormats(instr);
                        break;
                    case "DATE_FORMAT":
                        date = tryParseWithDateFormats(instr);
                        break;
                    default:
                        logger.error("Invalid DateTimeFormat entry: {}", entry);
                }
                if (date != null) {
                    break;
                }
            }
            if (date == null) {
                try {
                    date = Instant.from(DateTimeFormatter.ISO_INSTANT.parse(instr)).atZone(GMT).toLocalDateTime();
                } catch (DateTimeParseException e) {
                    // ignore
                }
            }
            // Use the default if required
        }
        if (date == null && supplyDefaultOnBad) {
            date = LocalDateTime.now();
        }
        return date;
    }

    private static String cleanDateString(String dateString) {
        String instr = dateString.replaceAll("\t+", " "); // tabs
        instr = instr.replaceAll("[ ]+", " "); // multiple spaces
        instr = instr.replaceAll("=0D$", ""); // common qp'ified ending
        return instr;
    }

Reply:
I understand the Legacy class will ultimately go away, but should it still have its own tests in the meantime?

Reply:
I can add the same tests that I added for the new class. Now that I think about it, it's probably a good idea to help assess the differences in results between the three classes.

Reply:
Attempted that and everything failed lol. Which makes sense because the number of formats the legacy class was designed can handle is much more limited. Tests the for legacy class will be handled elsewhere.

Reply:
> Attempted that and everything failed lol. Which makes sense because the number of formats the legacy class was designed can handle is much more limited. Tests the for legacy class will be handled elsewhere.

Sounds good!

Reply:
For the comment about potentially refactoring wrt the null checks, I spent a little time considering solutions. I refactored so some of the null checks were no longer necessary, and tried using return statements in some places. Let me know what you think, I was struggling to find the cleanest way of doing this.

Reply:
I like this!

Reply:
Here's a suggestion. I'm sure there is a way to condense this via lambda, but this is fairly readable.
```
            List<Function<String, LocalDateTime>> methodList = new ArrayList();
            methodList.add(date -> tryParseWithDateTimeZoneFormats(date));
            methodList.add(date -> tryParseWithDateTimeOffsetFormats(date));
            methodList.add(date -> tryParseWithDateTimeFormats(date));
            methodList.add(date -> tryParseWithDateFormats(date));

            for (Function<String, LocalDateTime> method : methodList) {
                LocalDateTime date = method.apply(instr);
                if (date != null) {
                    return date;
                }
            }
```